### DISCUSSION TITLE
DataGrid - OnCellPointerReleased ?

### DISCUSSION THREAD

**Discussion by KrzysztofDusko** [AvaloniaUI/Avalonia#14166] [Q&A] [answered]
We have **CellPointerPressed** and **PointerReleased** but ther is no **CellPointerReleased** . There is any way do determine DataGridCell under pointer ? I want to implement "lite version" of selection by mouse gesture, idea is to start selection by mouse click and ends with mouse release . I know about tree data grid but i have to use DataGrid for now. maybe I can calculate cell by e.GetPosition(firstClickedCell)?

this is right Way  ? 

```c#

        PointerReleasedEvent.AddClassHandler<DataGridCell>(
    (x, e) =>
    {
        // do some job
    }    , handledEventsToo: true);
```

**Comment by timunie** [comment, marked as answer]
e.Source should give you the original control clicked. Use VisualTreeExtensions to search for parent Cell.

**Reply by KrzysztofDusko** [reply]
thank you for your answer. This works.. but if i change backgroud of cell in this event handler this way
``` x.Background = Brushes.Orange;```
ui have 
![image](https://github.com/AvaloniaUI/Avalonia/assets/69449360/77eba226-1fc3-40ac-b7a4-52f80116a763)

but if i scroll down and up i have this.. , sometime sothing different.. state of cell is not saved.. (some recycling is used in data grid)
![image](https://github.com/AvaloniaUI/Avalonia/assets/69449360/9b677c71-3555-4076-8a40-609ff7c39242)

Is it possible to change cell's backgound in persitent manner ? 

I tried with style selector.. but DataContext returnet to converter is data context of entire row.. (in my use case data grid is created dynamicly, column count in unkown - data context of row is object array)
```xml
<Style Selector="DataGridCell.class0">
   <Setter Property="Background" Value="{Binding ., Converter={StaticResource converter}}"  />
</Style>
```

**Reply by timunie** [reply]
No it's virtualized so control will be lost during scrolling. This is by design. You need a way for a converter that knows data.

**Reply by KrzysztofDusko** [reply]
but data binding one cell to one value (not to whole row) in only possible if DataGridTemplateColumn is used ?

**Reply by timunie** [reply]
Probably a MultiConverter can be used. Or you add the classes in the `AutoGeneratingColumn` event

```cs
        private void MyDataGrid_OnAutoGeneratingColumn(object? sender, DataGridAutoGeneratingColumnEventArgs e)
{
    if (e.Column.Header.ToString() == "Name")
    {
        e.Column.CellStyleClasses.Add("MyClass");
    }
}
```
